example/test/settings_view_test.dart

```diff
@@ -9,11 +11,44 @@ void main() {
 
   /// Generate golden.
   testWidget<SettingsView>(
-    desc: 'SettingsScreen',
+    desc: 'SettingsScreen - goldens',
     widgetBuilder: (context, _) => widget.build(context),
     setup: (context, data) {
       when(() => context.watch<SettingsController>().themeMode)
           .thenReturn(data);
     },
   );
+
+  /// Widget test for DropdownButton.
+  testWidget<SettingsView>(
+      desc: 'SettingsScreen - DropdownButton test',
+      widgetBuilder: (context, _) => widget.build(context),
+      setup: (context, data) {
+        when(() => context.watch<SettingsController>().themeMode)
+            .thenReturn(data);
+        when(() => context
+                .read<SettingsController>()
+                .updateThemeMode(ThemeMode.dark))
+            .thenAnswer((invocation) => Future.value());
+      },
+      test: (tester, _) async {
+        /// A way to get context inside test.
+        Builder(builder: (context) {
```

Comment:
I think we can pass `BuildContext` to `test` function to avoid using of `Builder`